vignettes/howtogenefilter.Rmd

```diff
@@ -0,0 +1,190 @@
+
+FIXME: need to replace this with something else
+Our last example is to select genes that are
+differentially expressed in at least one of the three groups defined
+by \verb+cov3+.
+To do that we use an Anova filter. This filter uses an analysis of
+variance appraoch (via the \verb+lm+) function to test the hypothesis
+that at least one of the three group means is different from the other
+two. The test is applied, then the $p$--value computed. We select
+%those genes that have a low $p$--value.
+
+
+Afilter <- Anova(eset$cov3)
+aff <- filterfun(Afilter)
+wh4 <- genefilter(exprs(eset), aff)
+sum(wh4)
+
+We see that there are 14 genes that pass this filter and that are
+candidates for further exploration.
+
+\section*{Selecting genes that appear useful for prediction}
+
+The function \texttt{knnCV} defined below performs$k$--nearest neighbour classification
+using leave--one--out cross--validation.
+At the same time it aggregates the genes that were selected. The
+function returns the predicted classifications as its returned
+value. However, there is an additional side effect. The number of
+times that each gene was used (provided it was at least one) are
+recorded and stored in the environment of the aggregator \verb+Agg+.
+These can subsequently be retrieved and used for other purposes.
+
+```{r}
+knnCV <- function(x, selectfun, cov, Agg, pselect = 0.01, scale=FALSE) {
+   nc <- ncol(x)
+   outvals <- rep(NA, nc)
+   for(i in seq_len(nc)) {
+      v1 <- x[,i]
+      expr <- x[,-i]
+      glist <- selectfun(expr, cov[-i], p=pselect)
+      expr <- expr[glist,]
+      if( scale ) {
+        expr <- scale(expr)
+        v1 <- as.vector(scale(v1[glist]))
+      }
+      else
+         v1 <- v1[glist]
+      out <- paste("iter ",i, " num genes= ", sum(glist), sep="")
+      print(out)
+      Aggregate(row.names(expr), Agg)
+      if( length(v1) == 1)
+         outvals[i] <- knn(expr, v1, cov[-i], k=5)
+      else
+          outvals[i] <- knn(t(expr), v1, cov[-i], k=5)
+    }
+    return(outvals)
+}
+```
+
+```{r}
+ gfun <- function(expr, cov, p=0.05) {
+    f2 <- ttest(cov, p=p)
+    ffun <- filterfun(f2)
+    which <- genefilter(expr, ffun)
+  }
+```
+
+Next we show how to use this function on the dataset
+\verb+geneData+.
+
+<<aggregate2, results=hide>>=
+  library("class")
+
+  ##scale the genes
+  ##genescale is a slightly more flexible "scale"
+  ##work on a subset -- for speed only
+  
+  geneData <- genescale(exprs(sample.ExpressionSet)[1:75,], 1)
+  Agg <- new("aggregator")
+  testcase <- knnCV(geneData, gfun, sample.ExpressionSet$type, Agg,pselect=0.05) 
+aggregate3 <- sort(sapply(aggenv(Agg), c),
+decreasing=TRUE) 
+
+The environment `Agg` contains, for each gene, the
+number of times it was selected in the cross-validation.
+
+# Session Information {#session-information .unnumbered}
+
+The version number of R and packages loaded for generating the vignette
+were:
+
+\<\<sessionInfo,echo=FALSE,results=tex\>\>= toLatex(sessionInfo()) @
```

Comment:
Please change this into an R Markdown code block. It can just have `sessionInfo()` in the code block, no need to add `toLatex` anymore since we are using R Markdown.